code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -0,0 +1,328 @@
+		add_language("Galactic Common")
+		default_language = all_languages["TerrorSpider"]
+
+		name += " ([rand(1, 1000)])"
+		msg_terrorspiders("[src] has grown in [get_area(src)].")
+		if(name_usealtnames)
+			name = pick(altnames)
+		if(is_away_level(z))
+			spider_awaymission = 1
+			ts_count_alive_awaymission++
+		else
+			ts_count_alive_station++
+		// after 30 seconds, assuming nobody took control of it yet, offer it to ghosts.
+		spawn(150) // deciseconds!
+			CheckFaction()
+		spawn(300) // deciseconds!
+			if(spider_awaymission)
+				return
+			else if(ckey)
+				notify_ghosts("[src] has appeared in [get_area(src)]. (already player-controlled)")
+			else if(ai_playercontrol_allowingeneral && ai_playercontrol_allowtype)
+				notify_ghosts("[src] has appeared in [get_area(src)].", enter_link = "<a href=?src=\ref[src];activate=1>(Click to control)</a>", source = src)
+
+/mob/living/simple_animal/hostile/poison/terror_spider/Destroy()
+	ts_spiderlist -= src
+	..()
+
+/mob/living/simple_animal/hostile/poison/terror_spider/Life()
+	if(stat != DEAD)
+		if(regen_points < regen_points_max)
+			regen_points += regen_points_per_tick
+		if((bruteloss > 0) || (fireloss > 0))
+			if(regen_points > regen_points_per_hp)
+				if(bruteloss > 0)
+					adjustBruteLoss(-1)
+					regen_points -= regen_points_per_hp
+				else if(fireloss > 0)
+					adjustFireLoss(-1)
+					regen_points -= regen_points_per_hp
+		if(prob(5))
+			CheckFaction()
+	else if(stat == DEAD)
+		if(prob(2))
+			// 2% chance every cycle to decompose
+			visible_message("<span class='notice'>\The dead body of the [src] decomposes!</span>")
+			gib()
+	..()
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/death(gibbed)
+	if(!gibbed)
+		msg_terrorspiders("[src] has died in [get_area(src)].")
+	if(!hasdroppedloot)
+		hasdroppedloot = 1
+		ts_count_dead++
+		ts_death_last = world.time
+		if(spider_awaymission)
+			ts_count_alive_awaymission--
+		else
+			ts_count_alive_station--
+	..()
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/spider_special_action()
+	// Do nothing, this proc only exists to be overriden
+
+/mob/living/simple_animal/hostile/poison/terror_spider/Bump(atom/A)
+	if(istype(A, /obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/L = A
+		if(L.density)
+			try_open_airlock(L)
+	if(istype(A, /obj/machinery/door/firedoor))
+		var/obj/machinery/door/firedoor/F = A
+		if(F.density && !F.blocked)
+			F.open()
+	..()
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/msg_terrorspiders(var/msgtext)
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/T in ts_spiderlist)
+		if(T.stat != DEAD)
+			to_chat(T, "<span class='terrorspider'>TerrorSense: " + msgtext + "</span>")
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/CheckFaction()
+	if(faction.len != 1 || (!("terrorspiders" in faction)) || master_commander != null)
+		visible_message("<span class='danger'>[src] writhes in pain!</span>")
+		death()
+
```

Comment:
You might want to do the `log_runtime` thing here, since this looks like a "Bad program state" thing